src/Schema/Provider/FromFilesSchemaProvider.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Cycle\Schema\Provider;
+
+use InvalidArgumentException;
+use LogicException;
+use Yiisoft\Aliases\Aliases;
+use Yiisoft\Yii\Cycle\Schema\SchemaProviderInterface;
+
+/**
+ * Be careful, using this class may be insecure.
+ */
+final class FromFilesSchemaProvider implements SchemaProviderInterface
+{
+    private array $files = [];
+    private Aliases $aliases;
+
+    public function __construct(Aliases $aliases)
+    {
+        $this->aliases = $aliases;
+    }
+
+    public function withConfig(array $config): self
+    {
+        if (empty($config['files'])) {
+            throw new InvalidArgumentException('Files not set.');
+        }
+
+        if (!is_array($config['files'])) {
+            throw new InvalidArgumentException('The "files" parameter must be an array.');
+        }
+
+        $files = $config['files'];
+
+        $files = array_map(
+            fn ($file) => $this->aliases->get($file),
+            $files
+        );
+
+        $new = clone $this;
+        $new->files = $files;
+        return $new;
+    }
+
+    public function read(): ?array
+    {
+        $schema = [];
+        foreach ($this->files as $file) {
+            if (is_file($file)) {
+                foreach (require $file as $role => $definition) {
+                    if (array_key_exists($role, $schema)) {
+                        throw new LogicException('The "' . $role . '" role already exists in the DB schema.');
+                    }
+                    $schema[$role] = $definition;
+                }
+            }
+        }
+        return empty($schema) ? null : $schema;
```

Comment:
Should we convert an empty array to the NULL if empty arrays are intentionally specified in the merged files?

Reply:
Difficult question.

I see two variants:

1) If read ALL files, then return schema, else `null`.
2) If read AT LEAST ONE file, then return schema, else `null`.

Reply:
Add `strict` parameter.